contrib/cirrus/lib.sh

```diff
@@ -151,16 +165,118 @@ install_ooe() {
 }
 
 showrun() {
-    if [[ "$1" == "--background" ]]
+    local context
+    context=($(caller 0))
+    echo '+ '$(printf " %q" "$@")"  # ${context[2]}:${context[0]} in ${context[1]}()" > /dev/stderr
```

Comment:
These echo-printfs make my head hurt; this one in particular is really confusing. Is there a specific reason for not using simple printf?
```bash
    printf "+ %q # %s:%s in %s()\n" "$@" ${context[2]} ${context[0]} ${context[1]} >/dev/stderr
```
(Likewise for all other echo-printf instances throughout).

Also btw you probably want `local -a` on line 168

Reply:
Yeah there was a reason, but damned if I can remember even if it was even a good one :smile:   Copy-pasting your suggestions above.  I confirmed `local -a` works on ubuntu 18's older bash.